pyqtorch/utils.py

```diff
@@ -5,11 +5,15 @@
 
 import torch
 
-from pyqtorch.matrices import DEFAULT_MATRIX_DTYPE
+from pyqtorch.matrices import DEFAULT_MATRIX_DTYPE, DEFAULT_REAL_DTYPE
 
 State = torch.Tensor
 Operator = torch.Tensor
 
+ATOL = 1e-07 if DEFAULT_REAL_DTYPE == torch.float32 else 1e-014
+RTOL = 1e-05 if DEFAULT_REAL_DTYPE == torch.float32 else 0.0
```

Comment:
Why do you need the `1e-05` rtol for float32 though? If tests are not passing otherwise, maybe it would be sufficient to make the default ATOL `1e-06` for that case. 

[According to this](https://en.wikipedia.org/wiki/Machine_epsilon) the machine epsilon for float32 can be anywhere from `5e-08` to `1e-07`, so probably we are pushing it too close with the current value.

Reply:
done